### PR TITLE
Compute index/position in Merkle tree correctly

### LINKED ISSUE #444: Indices in Message Recover Command are incorecly interpreted as positions in tree
## Current Specifications

In the verification of the Message Recovery Command, the first and the last entry of `idxs` are interpreted as the positions of the correponding CCMx and in outbox tree:
```python
    # Check that the CCMs are still pending. We check only the first one,
    # as the idxs are sorted in ascending order.
    if trsParams.idxs[0] < terminatedOutboxAccount(trsParams.chainID).partnerChainInboxSize:
        raise Exception("Cross-chain message is not pending.")

    # Check that the CCM indices do not exceed the outbox size. We check only the last one,
    # as the idxs are sorted in ascending order.
    if terminatedOutboxAccount(trsParams.chainID).outboxSize <= trsParams.idxs[-1]:
        raise Exception("Cross-chain message was never in the outbox.")
```
However, the indices in `idxs` have a leading 1 in their bit representation as specified in [LIP 0031](https://github.com/LiskHQ/lips/blob/1e27a5ba99a55d71e6d83f44b6e21f18dda98741/proposals/lip-0031.md?plain=1#L194). Hence, they cannot directly be interpreted as the positions in the outbox tree.

## Proposed Specifications

Remove the leading 1 in the bit representation before interpreting the enties of `idxs` as positions in the outbox tree.

## Affected LIPs

0054

### PR DESCRIPTION
**Pull request by AndreasKendziorra**
Fixes #444

### CODE REVIEW

**Review by ricott1 — APPROVED** [review]

**Review by gkoumout — APPROVED** [review]

**Review by sergeyshemyakov — APPROVED** [review]
Looks good

**Review by mitsuaki-u — APPROVED** [review]

File: proposals/lip-0054.md [hunk 1]
```diff
@@ -387,13 +387,21 @@ def verify(trs: Transaction) -> None:
             raise Exception("Cross-chain message indexes are not strictly increasing.")
     
     # Check that the CCMs are still pending. We check only the first one,
-    # as the idxs are sorted in ascending order.
-    if trsParams.idxs[0] < terminatedOutboxAccount(trsParams.chainID).partnerChainInboxSize:
+    # as the idxs are sorted in ascending order. Note that one must unset the most significant
+    # bit of an encoded index in idxs in order to get the position in the tree. To do this
+    # we must ensure that there are at least two bits, i.e. the value must be larger than 1.
+    # See https://github.com/LiskHQ/lips/blob/main/proposals/lip-0031.md#proof-serialization.
+    if idxs[0] > 1:
```
Comment (sergeyshemyakov) [inline proposals/lip-0054.md:394]: I think it should be `if idxs[0] <= 1`.

**Review by sergeyshemyakov — CHANGES_REQUESTED** [review]

File: proposals/lip-0054.md (hunk 1 above)
Comment (AndreasKendziorra) [reply proposals/lip-0054.md]: right :) I fixed it

**Review by sergeyshemyakov — APPROVED** [review]
Looks good

**Review by gkoumout — APPROVED** [review]

**Review by mitsuaki-u — APPROVED** [review]

**Review by janhack — APPROVED** [review]
Editorial approval

**Pull request merged**